File: 02Fabryka/Students/kkarkucinski/Factory/PhoneFactory.java
package Factory;

/**
 * Created by K.Karkucinski on 2016-12-15.
 */
public abstract class PhoneFactory {
    public Phone orderPhone(String type){
        Phone phone;
        phone = makePhone(type);
        phone.wrapping();
        phone.sending();
        return phone;
    }
    abstract Phone makePhone(String type);
}

File: 02Fabryka/Students/2016/mbowzyk/src/main/kotlin/fabricMethod/PhoneFactory.kt
package fabric.fabricMethod

import fabric.fabricMethod.models.Phone

abstract class PhoneFactory {

    abstract protected fun buildPhone(phoneModel: PhoneModel) : Phone

    fun orderPhone(phoneModel: PhoneModel) : Phone {
        val phone: Phone = buildPhone(phoneModel)
        println("creating phone ${phone.toString()}")
        phone.prepare().building().packing()
        println("phone is ready ${phone.toString()}")

        return phone
    }

}